bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/embed/OEmbedClientImpl.java

```diff
@@ -106,9 +111,19 @@ public OEmbedResponse getResponse(String url) {
         } else if (jaxbContext != null && OEmbedResponse.Format.XML == OEmbedResponse.Format.fromString(config.format())) {
             try {
                 String xmlURL = buildURL(config.endpoint(), url, OEmbedResponse.Format.XML.getValue(), null, null);
+                InputStream xmlStream = getData(xmlURL);
+
+                //Disable XXE
+                SAXParserFactory spf = SAXParserFactory.newInstance();
+                spf.setFeature("http://xml.org/sax/features/external-general-entities", false);
+                spf.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+                spf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+
+                //Do unmarshall operation
+                Source xmlSource = new SAXSource(spf.newSAXParser().getXMLReader(), new InputSource(xmlStream));
```

Comment:
try..with..resources on both xmlreader and xmlstream as otherwise they might not be properly closed.

Reply:
@kwin `spf.newSAXParser().getXMLReader()` is not a Closable, it cannot be used/closed with `try with resources`

Reply:
You are right, I have now seen it as well. Looks good now to me.